src/components/Settings/Settings.tsx

```diff
@@ -171,113 +201,151 @@ function SettingsContent({
         }
     });
 
-    const renderPageContent = () => {
-        if (!activePage) {
+    React.useEffect(() => {
+        if (!selectionInitialPage) return;
+        setCurrentPage(selectionInitialPage);
+    }, [selectionInitialPage]);
+
+    React.useEffect(() => {
+        if (!selected.selectedRef?.current) return;
+
+        selected.selectedRef.current.scrollIntoView();
+    }, [selected.selectedRef]);
```

Comment:
In what case can the value of `selected.selectedRef` change?

Reply:
1. We change the `selection` property.
2. We change the page and the selected setting is not rendered anymore (or vice versa).